### PR TITLE
aws-iot-securetunneling-localproxy: init at 2.0.1

### PR DESCRIPTION
**Pull request by anton-dessiatov**
###### Motivation for this change

Using AWS IoT infrastructure for [secure tunneling](https://docs.aws.amazon.com/iot/latest/developerguide/local-proxy.html) requires an application to forward connections from local machine to IoT device in the fleet. This PR packages AWS sample application for doing so.

This application is pretty niche, but it was (and still is) useful for me so maybe someone else could benefit from it as well.

Package name is a bit mouthful but I've borrowed that from GitHub source repository name, so I guess it's okay to keep it to reduce ambiguity.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner  = "aws-samples";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "sha256-WGAzf66TaDpzqbdDzP8wyaC89G8FOH0tgR4to+QtYEU=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost protobuf openssl catch2 ];
+  patches = [ ./cmake.patch ];
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix:16]:
```suggestion
```

File: pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+stdenv.mkDerivation rec {
+  pname = "aws-iot-securetunneling-localproxy";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner  = "aws-samples";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "sha256-WGAzf66TaDpzqbdDzP8wyaC89G8FOH0tgR4to+QtYEU=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix:12]:
```suggestion
  };

  patches = [ ./cmake.patch ];
```

File: pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ lib, stdenv, fetchFromGitHub, cmake, boost, protobuf, openssl, catch2 }:
+
+stdenv.mkDerivation rec {
+  pname = "aws-iot-securetunneling-localproxy";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner  = "aws-samples";
+    repo   = pname;
+    rev    = "v${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix:10]:
```suggestion
    owner = "aws-samples";
    repo = pname;
    rev = "v${version}";
```
our formatting tools do not suppor this.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137163` at c5a54d5b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>aws-iot-securetunneling-localproxy</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/tools/virtualization/aws-iot-securetunneling-localproxy/default.nix:16:15:
```
   |
16 |   patches = [ ./cmake.patch ];
   |               ^
```
</li>
  </ul>
</details>

**Comment by anton-dessiatov** [comment]
Thanks for the review. I've accepted suggestions and added comment that describes a purpose of the patch.

**Comment by drupol** [comment]
Superseded by #245073

**Pull request closed without merging**